server.go

```diff
@@ -299,15 +463,30 @@ func (s *Server) serveConn(newConn net.Conn) {
 	}
 	s.connections[frisbeeConn] = struct{}{}
 	s.connectionsMu.Unlock()
-
-	err = s.handlePacket(frisbeeConn, connCtx)
-	_ = frisbeeConn.Close()
-	s.onClosed(frisbeeConn, err)
-	s.connectionsMu.Lock()
-	if !s.shutdown.Load() {
-		delete(s.connections, frisbeeConn)
+	if s.concurrency == 0 {
+		s.handleUnlimitedPacket(frisbeeConn, connCtx)
+		s.connectionsMu.Lock()
```

Comment:
This tidy up can come out of the if block? It's identical in all 3 cases?
(The check for shutdown -> delete(s.connections, frisbeeConn))

Reply:
Makes sense!